controllers/statusreport/statusreport_adapter.go

```diff
@@ -247,17 +248,24 @@ func (a *Adapter) EnsureSnapshotFinishedAllTests() (controller.OperationResult,
 // list of integrationTestScenarios.
 func (a *Adapter) determineIfAllIntegrationTestsFinishedAndPassed(integrationTestScenarios *[]v1beta2.IntegrationTestScenario, testStatuses *intgteststat.SnapshotIntegrationTestStatuses) (bool, bool) {
 	allIntegrationTestsFinished, allIntegrationTestsPassed := true, true
+	integrationTestsFinished := len(*integrationTestScenarios)
+	integrationTestsPassed := len(*integrationTestScenarios)
+
 	for _, integrationTestScenario := range *integrationTestScenarios {
 		integrationTestScenario := integrationTestScenario // G601
 		testDetails, ok := testStatuses.GetScenarioStatus(integrationTestScenario.Name)
 		if !ok || !testDetails.Status.IsFinal() {
 			allIntegrationTestsFinished = false
+			integrationTestsFinished--
```

Comment:
Both counters are placed logically in wrong places, they must be in else branches

Reply:
nevermind, they are fine but confusing, negation and decrement combined